content/shaders/World.bsl

```diff
@@ -0,0 +1,169 @@
+#include "$ENGINE$\GBufferOutput.bslinc"
+#include "$ENGINE$\PerCameraData.bslinc"
+#include "$ENGINE$\PerObjectData.bslinc"
+#include "$ENGINE$\VertexInput.bslinc"
+
+shader Surface
+{
+  mixin GBufferOutput;
+  mixin PerCameraData;
+  mixin PerObjectData;
+  mixin VertexInput;
+
+  code
+  {
+    [alias(gAlbedoTex)] SamplerState gAlbedoSamp;
+
+    Texture2D gAlbedoTex = white;
+
+    struct ZenVertexInput
+    {
+      #if POSITION_2D
```

Comment:
Don't you need a `variation` to do this or is it not yet used?

Reply:
I don't know, this was copied from the original vertex input structure. I haven't read that deep into the shader language yet. However, I've seen some variations being defined in one of the included headers.